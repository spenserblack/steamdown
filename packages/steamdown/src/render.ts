import type { fromMarkdown } from "mdast-util-from-markdown";

export type Root = ReturnType<typeof fromMarkdown>;

export function render(root: Root): string {
  const { data } = root;
  if (data == null) {
    return "";
  }
  return JSON.stringify(data);
}
