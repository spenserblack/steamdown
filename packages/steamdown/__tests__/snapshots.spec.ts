import { readFile } from 'fs/promises';
import { resolve } from 'path';
import { describe, test, expect } from '@jest/globals';
import render from '../src/index';

describe('parse spec', () => {
  const tests = ['basic'].map((name) => [name]);
  test.each(tests)('%s', async (name: string) => {
    const path = resolve(__dirname, './input', `${name}.md`);
    const src = await readFile(path, 'utf8');
    expect(render(src)).toMatchSnapshot();
  });
});
