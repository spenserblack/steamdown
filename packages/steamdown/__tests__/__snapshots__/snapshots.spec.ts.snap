// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse spec basic 1`] = `
"[h1]Heading 1[/h1]

[h2]Heading 2[/h2]

[h3]Heading 3[/h3]

[h4]Heading 4[/h4]

[h5]Heading 5[/h5]

[h6]Heading 6[/h6]

This is [b]bold[/b] text

This is [u]underlined[/u] text

This is [i]italicized[/i] text
This is also [i]italicized[/i] text

This is [strike]strikethrough[/strike] text

This is [spoiler]spoiler[/spoiler] text

This is
text with a break

This text [noparse][b]is[/b] *literal*[/noparse]

[noparse]
This is [i]also[/i]
literal __text__.
[/noparse]

[hr][/hr]

This is a link to [url=https://store.steampowered.com/]Steam[/url]!

An image is also [url=https://simpleicons.org/icons/steam.svg]a link[/url].

[list]
    [*]Bulleted list
    [*]Bulleted list
    [*]Bulleted list
[/list]

[olist]
    [*]Ordered list
    [*]Ordered list
    [*]Ordered list
[/olist]

[quote]this is a quote without an author[/quote]

[quote=author]this is a quote with an author[/quote]

This is [code]inline[/code] code

[code]
This is a code block
[/code]

[table]
  [tr]
    [th]This is[/th]
    [th]a table[/th]
  [/tr]
  [tr]
    [td]That's[/td]
    [td]cool![/td]
  [/tr]
[/table]

newlines


are preserved"
`;

exports[`parse spec list-items 1`] = `
"[list]
    [*][url=https://example.com]link[/url]
    [*][i]emphasis[/i]
    [*][b]strong[/b]
[/list]"
`;
